FHEM/60_uba.pm

```diff
@@ -550,187 +240,266 @@ sub uba_Get($@) {
+      $reading_datetime =~ s/ 24:/ 00:/;
+      Log3 $name, 4, "-0 DateTime (after fix): ".$reading_datetime;
+      my $tp = Time::Piece->strptime($reading_datetime, "%Y-%m-%d %H:%M:%S %z");
+      $measured_datetime_timestamp = int($tp->epoch+(23*60*60));
+      Log3 $name, 4, "-0 DateTime (after fix) (from timestamp): ".localtime($measured_datetime_timestamp)->strftime("%Y-%m-%d %H:%M:%S");
+    } else {
+      my $tp = Time::Piece->strptime($reading_datetime, "%Y-%m-%d %H:%M:%S %z");
+      $measured_datetime_timestamp = $tp->epoch;
+    }
+
+    # Debug
+    Log3 $name, 4, "-0 DateTime (timestamp): ".$measured_datetime_timestamp;
+
+    # Wann wurde das letzte Update an Daten eingespielt?
+    my $lastupdate_timestamp = ReadingsVal( $name, ".lastUpdate", $hash->{helper}{initial_lastupdate});
+    Log3 $name, 4, "-0 LastUpdate (timestamp): ".$lastupdate_timestamp;
+
+    # Skip duplicate
+    if($measured_datetime_timestamp <= $lastupdate_timestamp) {
+      Log3 $name, 4, "Skip this reading. It should be already in database.";
+      next;
+    }
+
+    # Start data update
     readingsBeginUpdate($hash);
-    $hash->{".updateTimestamp"} = FmtDateTime($time);
-    readingsBulkUpdate( $hash, $param->{type}, $value );
-    $hash->{CHANGETIME}[0] = FmtDateTime($time);
-    readingsEndUpdate($hash,1);
 
-  }
+    # Set specific reading time from dataset
+    $hash->{".updateTimestamp"} = FmtDateTime($measured_datetime_timestamp);
+    my $changeindex = 0;
 
-  Log3 $name, 5, "JSON data for ".$param->{type}."\n".Dumper($json);
+    # Save Luftqualitätsindex
+    readingsBulkUpdateIfChanged($hash,"luftqualitaetsindex",$json->{"data"}->{$station}->{$datetimekey}[1]);
+    $hash->{CHANGETIME}[$changeindex++] = FmtDateTime($measured_datetime_timestamp);
 
-  if($received > 0)
-  {
-    Log3 $name, 3, "Received $received values for ".$param->{type}." from after ".FmtDateTime($param->{range}+1800);
-    readingsSingleUpdate( $hash, ".lastUpdate".$param->{type}, $lastdata, 0 );
-    readingsSingleUpdate( $hash, "lastUpdate".$param->{type}, FmtDateTime($lastdata), 0 ) if(AttrVal($name, "showTimeReadings", 0) eq 1);
-  } else {
-    my $twentyfour = int(time)-(24*60*60);
-    readingsSingleUpdate( $hash, ".lastUpdate".$param->{type}, $twentyfour, 0 ) if($lastdata < $twentyfour);
-    Log3 $name, 2, "Skipping missing readings before ".FmtDateTime($twentyfour) if($lastdata < $twentyfour);
-  }
+    # Save Luftqualitätsindex_name
+    readingsBulkUpdateIfChanged($hash,"luftqualitaetsindex_name",$airquality{$json->{"data"}->{$station}->{$datetimekey}[1]});
+    $hash->{CHANGETIME}[$changeindex++] = FmtDateTime($measured_datetime_timestamp);
+  
+    # Search and work on each specific available dataset
+    my $dataset_size = @{$json->{"data"}->{$station}->{$datetimekey}};
+    Log3 $name, 5, "x dataset_size: $dataset_size";
+    for ( my $i = 3; $i < $dataset_size; $i++ ) {
+      Log3 $name, 4, "-$i 0 component ID: ".$i;
```

Comment:
Ich bin ganz davon weg, `Log3` für sowas zu verwenden. Ich hab mir in meinen eigenen Namespace eine Debugging-Funktion gelegt, die einerseits `Debug()` nutzt, andererseits aber ein Debug-Attribut berücksichtigt. So kannst du dann auch in Loglevel 1 loggen ohne dass du das Log des Users vollmüllst, wenn er ein niedriges Loglevel eingestellt hat.